cfgov/search/elasticsearch_helpers.py

```diff
@@ -181,22 +181,28 @@ class WagtailSignalProcessor(BaseSignalProcessor):
     Wagtail events and calls the appropriate signals to update the search
     index.
 
-    It also translates the `instance` object that gets passed by Wagtail
-    signal handlers into its `instance.specific` equivalent for any
-    downstream logic that depends on the specific page type.
+    It also translates the `instance` Page object that gets passed by Wagtail
+    signal handlers into its AbstractFilterPage equivalent (if it so inherits)
+    to ensure that the filterable page search index is properly updated.
     """
 
-    def handle_delete(self, sender, instance, **kwargs):
-        if isinstance(instance, Page):
-            instance = instance.specific
+    def check_afp(self, instance):
+        # If the provided instance is a Wagtail page instance that inherits
+        # from AbstractFilterPage, convert it to an AFP instance.
+        from v1.models import AbstractFilterPage
 
-        super().handle_delete(sender, instance, **kwargs)
+        if isinstance(instance, Page) and issubclass(
+            instance.specific_class, AbstractFilterPage
+        ):
+            instance = AbstractFilterPage.objects.get(pk=instance.pk)
```

Comment:
In theory we could save a database lookup here if the instance is already a more-specific page type, but I don't think we can assume that `Page.save()` or `Page.delete()` will always be called on the specific instance type.

Reply:
One question, isn't `isinstance(instance, Page)` implied already by `issubclass(..., AbstractFilterPage)`? Is there a reason to test both?

Reply:
Oh, sorry, I see now. Check if it's a `Page` to know if it has `specific_class`.

Reply:
I don't think I would worry too much about extra database queries here, personally. This is only called on `save()` and `delete()`, it looks like, for which performance optimization is nice, but not as important as the basic GET render cycle, IMHO.